### ISSUE TITLE
Error while downloading module from terraform registry

### ISSUE THREAD

**Issue by eduardohf-ciandt** [hashicorp/terraform#23833]
Error:

:        Error: Failed to download module
:
:        Could not download module "service_accounts" (../../../main.tf:6) source code
:        from
:        "https://api.github.com/repos/terraform-google-modules/terraform-google-service-accounts/tarball/v2.0.2//*?archive=tar.gz":
:        Error opening a gzip reader for /tmp/tmp.pBync0kChF/getter826662469/archive:
:        EOF.

Code:

module "service_accounts" {
  source        = "terraform-google-modules/service-accounts/google"
  version       = "~> 2.0"
  project_id    = var.project_id
  prefix        = "sa"
  names         = [var.name]
  generate_keys = var.generate_key
  project_roles = local.project_roles
}

**Comment by n0rig** [comment]
Hi, I am seeing the same issue as well.

It looks like terraform gets the correct working URL path:
```
2020/01/10 09:53:07 [DEBUG] looking up module location from "https://registry.terraform.io/v1/modules/terraform-aws-modules/vpc/aws/2.15.0/download"
2020/01/10 09:53:07 [TRACE] HTTP client GET request to https://registry.terraform.io/v1/modules/terraform-aws-modules/vpc/aws/2.15.0/download
2020/01/10 09:53:07 [TRACE] ModuleInstaller: slr-vpc.vpc terraform-aws-modules/vpc/aws 2.15.0 is available at "https://api.github.com/repos/terraform-aws-modules/terraform-aws-vpc/tarball/v2.15.0//*?archive=tar.gz"
2020/01/10 09:53:07 [DEBUG] will download "https://api.github.com/repos/terraform-aws-modules/terraform-aws-vpc/tarball/v2.15.0?archive=tar.gz" to .terraform/modules/slr-vpc.vpc
```

But, then, it somehow get's converted to a path that does not work later on:
```
Could not download module "vpc"
(.terraform/modules/slr-vpc/slr-vpc/aws_vpc.tf:9) source code from
"https://api.github.com/repos/terraform-aws-modules/terraform-aws-vpc/tarball/v2.15.0//*?archive=tar.gz":
```

**Comment by jbardin** [comment]
Duplicate of #23721 and #23829

**Comment by justincampbell** [comment]
This should now be fixed! Details in #23834
